sympy/integrals/prde.py

```diff
@@ -344,7 +344,7 @@ def prde_no_cancel_b_large(b, Q, n, DE):
 
     if all(qi.is_zero for qi in Q):
         dc = -1
-        M = zeros(0, 2, DE.t)
+        M = zeros(0, m, DE.t)
     else:
         dc = max([qi.degree(DE.t) for qi in Q])
         M = Matrix(dc + 1, m, lambda i, j: Q[j].nth(i), DE.t)
```

Comment:
Not really sure why I ever did it like this since it isn't this way in the book, but this line can just be used for both cases. I think I just didn't have a clear understanding of how empty matrices work, or possibly there was some bug with how they behaved back then, I don't know.

This is page 234 in Bronstein's book, ParamPolyRishDENoCancel1, for anyone checking. Really these docstrings ought to reference the exact pseudocode + page number in the book to make this easier.

Reply:
Maybe you did it like this because:
```
(Pdb) p Q[0].degree(DE.t)
-oo
```

Reply:
Oh to be clear, the dc = -1 part is there because that's how it's written in the book (and the ambiguity of degree(0) is clearly the reason). I meant I don't know why I put the matrix definition inside of the `if`. In the book, it's separate.

Reply:
Okay, I see what you mean. I've moved the matrix definition out.